ckan/lib/search/query.py

```diff
@@ -302,8 +302,13 @@ def run(self, query, permission_labels=None, **kwargs):
         '''
         assert isinstance(query, (dict, MultiDict))
         # check that query keys are valid
-        if not set(query.keys()) <= VALID_SOLR_PARAMETERS:
-            invalid_params = [s for s in set(query.keys()) - VALID_SOLR_PARAMETERS]
+        valid_solr_parameters = VALID_SOLR_PARAMETERS
+        for item in plugins.PluginImplementations(plugins.IPackageController):
+            if 'update_valid_solr_parameters' in dir(item):
```

Comment:
This is only needed because I haven't added fallback implementation to IPackageController yet